crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java

```diff
@@ -885,9 +890,16 @@ private ObjectNode buildArraySchema(ApiVersion crApiVersion, Property property,
         return result;
     }
 
-    private boolean isStringStringMap(PropertyType propertyType) {
+    /**
+     * Utility method to check if Map key-value pair match specific types.
+     * @param propertyType property to check
+     * @param keyType key Class
+     * @param valueType value Class
+     * @return true if key-value types are equal to specified types, false otherwise.
+     */
+    private boolean isSameTypesMap(PropertyType propertyType, Class<?> keyType, Class<?> valueType) {
```

Comment:
I don't think the name makes sense as the types are not the same. You have different types for key and value. So you should probably find a better name? Maybe `isMapWithTypes`?

Reply:
renamed to `isMapOfTypes`